crates/core/plugin_sm/src/operation_logs.rs

```diff
@@ -75,36 +84,57 @@ impl OperationLogs {
-                    list_logs.push(Reverse(path.to_string()));
+                if path.starts_with("software-list") {
+                    log_tracker
+                        .entry("software-list".to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
+                } else if path.starts_with("software-update") {
+                    log_tracker
+                        .entry("software-update".to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
+                } else {
+                    let file_name = path
+                        .split('-')
+                        .next()
+                        .ok_or(OperationLogsError::NoOperation)?;
+                    log_tracker
+                        .entry(file_name.to_string())
+                        .or_insert_with(BinaryHeap::new)
+                        .push(Reverse(path.to_string()));
                 }
             }
         }
 
-        while update_logs.len() > self.max_update_logs {
-            if let Some(rname) = update_logs.pop() {
-                let name = rname.0;
-                let path = self.log_dir.join(name.clone());
-                if let Err(err) = std::fs::remove_file(&path) {
-                    log::warn!("Fail to remove out-dated log file {} : {}", name, err);
+        for (key, value) in log_tracker.iter_mut() {
+            if key.starts_with("software-list") {
+                // only allow one update list file in logs
+                while value.len() > 1 {
+                    if let Some(rname) = value.pop() {
+                        let name = rname.0;
+                        let path = self.log_dir.join(name.clone());
+                        if let Err(err) = std::fs::remove_file(&path) {
+                            log::warn!("Fail to remove out-dated log file {} : {}", name, err);
+                        }
+                    }
                 }
-            }
-        }
-
-        while list_logs.len() > self.max_list_logs {
-            if let Some(rname) = list_logs.pop() {
-                let name = rname.0;
-                let path = self.log_dir.join(name.clone());
-                if let Err(err) = std::fs::remove_file(&path) {
-                    log::warn!("Fail to remove out-dated log file {} : {}", name, err);
+            } else {
+                // allow most recent five
+                while value.len() > 5 {
+                    if let Some(rname) = value.pop() {
+                        let name = rname.0;
+                        let path = self.log_dir.join(name.clone());
+                        if let Err(err) = std::fs::remove_file(&path) {
+                            log::warn!("Fail to remove out-dated log file {} : {}", name, err);
+                        }
+                    }
```

Comment:
This logic seem to be repeating multiple times. Can be refactored into a single method and reused.